### PR TITLE
[EC-260] Fundamental Terraform Core setup for Italy North region

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

This PR contains a definition of new resources to be deployed in Italy North region.

They are *completely separated* from West Europe region resources, in order to break the coupling between the two.
They use the modules-approach, to simplify resource searching, breaking coupling and get rid of the long file list. 

This core folder is intended to be used especially by cloud engineers, NOT developers; they instead shall define resources in their mono repos/application repos/domain folder, according to their patterns. Core folder shall contain fundamental blocks and nothing else. [An RFC](https://pagopa.atlassian.net/wiki/spaces/ENG/pages/910492147/RFC-ENG+Riorganizzazione+configurazioni+Terraform) with more details and explanations have been opened some times ago.

This PR also uses [HashiCorp module](https://github.com/hashicorp/terraform-cidr-subnets) to auto generate CIDR blocks for subnets. This is useful to avoid CIDR collision in concurrent PRs, especially in "distributed repositories", and is intended to be used also in all Terraform configuration which are going to interact with North Italy region.

Also naming conventions have been updated: location is introduced, as well as the instance number. The former is useful to distinguish resources based on location (pattern in part already adopted in West Europe but then abandoned); the latter is useful for migrations, in order to be more ripetitive than creative (aka inventing names to create unique resources similar to old ones and creating confusion). This approach is also [suggested by Microsoft](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming).

** WHAT IS STILL MISSING **
Remove resources from West Europe core state
Identity secrets in repository
Waiting for dx-typescript workflow to be merged

All will be added to THIS PR as soon as possible

### Motivation and context

Getting the new region opportunity to fix some historical issues.
We need to deploy first resources in Italy *very very very soon (this week)*

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/core/prod/italynorth/locals.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+locals {
+  prefix                   = "io"
+  env_short                = "p"
+  location_short           = "in"
```
Comment (pasqualedevita) [inline src/core/prod/italynorth/locals.tf:4]: itn

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/core/prod/italynorth/locals.tf (hunk 1 above)
Comment (pasqualedevita) [reply src/core/prod/italynorth/locals.tf:4]: for location short

**Comment by Krusty93** [comment]
I would like also to integrate [this](https://registry.terraform.io/providers/aztfmod/azurecaf/latest/docs/resources/azurecaf_name)

**Comment by Krusty93** [comment]
> I would like also to integrate [this](https://registry.terraform.io/providers/aztfmod/azurecaf/latest/docs/resources/azurecaf_name)

Discarded because of lack of support

**Pull request merged**